core/java/src/org/openda/exchange/dataobjects/NetcdfFileConcatenater.java

```diff
@@ -55,111 +68,261 @@ public static void main(String[] arguments) {
-						netcdfFileWriter.write(entry.getKey(), entry.getValue());
+					netcdfToAdd = NetcdfFile.open(netcdfFileToBeAdded.getAbsolutePath());
+					Dimension time = netcdfToAdd.findDimension("time");
+					if (time != null && time.isUnlimited()) {
+						BBUtils.copyFile(netcdfFileToBeAdded, targetNetcdfFile);
+					} else {
+						rewriteNetcdfFile(targetNetcdfFile, netcdfToAdd);
 					}
-				} catch (InvalidRangeException e) {
-					throw new RuntimeException("Error rewriting variables", e);
+				} finally {
+					if (netcdfToAdd != null) netcdfToAdd.close();
 				}
-				netcdfFileWriter.flush();
-				netcdfFileWriter.close();
+			} else {
+				concatenateNetcdfFiles(useOldValueOnOverlap, netcdfFileToBeAdded, targetNetcdfFile);
 			}
 		} catch (IOException e) {
-			throw new RuntimeException("Error", e);
+			throw new RuntimeException(e.getMessage(), e);
 		}
 	}
 
-	private static void addConcatenatedValueArraysToMaps(Map<Variable, Array> variableArraysMap, Map<Variable, Array> timeVariableArraysMap, Variable targetVariable, Variable timeVariableTarget, boolean concatenateTimeVariable, int targetLocationDimensionLength, double[][] targetValues, double[][] addedValues, double[] timesTarget, double[] convertedTimesToBeAdded, int totalTimesCombined, boolean firstAddedTimeOverlapping) {
+	private static void concatenateNetcdfFiles(boolean useOldValueOnOverlap, File netcdfFileToBeAdded, File targetNetcdfFile) throws IOException {
+		NetcdfFile sourceNetcdfFile = null;
+		NetcdfFileWriter netcdfFileWriter = null;
+		try {
+			sourceNetcdfFile = NetcdfFile.open(netcdfFileToBeAdded.getAbsolutePath());
+			List<Variable> variablesToBeAdded = sourceNetcdfFile.getVariables();
+			netcdfFileWriter = NetcdfFileWriter.openExisting(targetNetcdfFile.getAbsolutePath());
+
+			Map<Variable, Array> variableArraysMap = new HashMap<>();
+			Map<Variable, Array> timeVariableArraysMap = new HashMap<>();
+
+			concatenateVariables(useOldValueOnOverlap, sourceNetcdfFile, variablesToBeAdded, netcdfFileWriter, variableArraysMap, timeVariableArraysMap);
+			try {
+				for (Map.Entry<Variable, Array> entry : timeVariableArraysMap.entrySet()) {
+					netcdfFileWriter.write(entry.getKey(), entry.getValue());
+				}
+				for (Map.Entry<Variable, Array> entry : variableArraysMap.entrySet()) {
+					netcdfFileWriter.write(entry.getKey(), entry.getValue());
+				}
+			} catch (InvalidRangeException e) {
+				throw new RuntimeException("Error rewriting variables", e);
+			}
+			netcdfFileWriter.flush();
+		} finally {
+			if (netcdfFileWriter != null) netcdfFileWriter.close();
+			if (sourceNetcdfFile != null) sourceNetcdfFile.close();
+		}
+	}
+
+	private static void concatenateVariables(boolean useOldValueOnOverlap, NetcdfFile sourceNetCdfFile, List<Variable> variablesToBeAdded, NetcdfFileWriter netcdfFileWriter, Map<Variable, Array> variableArraysMap, Map<Variable, Array> timeVariableArraysMap) throws IOException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'concatenateVariables(boolean, NetcdfFile, List, NetcdfFileWriter, Map, Map)' has an NPath complexity of 4097](https://app.codacy.com/gh/OpenDA-Association/OpenDA/pullRequest?prid=5271142)